lib/glific/triggers.ex

```diff
@@ -154,19 +154,34 @@ defmodule Glific.Triggers do
 
   @spec create_trigger(map()) :: {:ok, Trigger.t()} | {:error, Ecto.Changeset.t()}
   def create_trigger(attrs) do
-    {:ok, flow} =
-      Repo.fetch_by(FlowRevision, %{flow_id: Map.get(attrs, :flow_id), status: "published"})
+    with {:ok, trigger} <-
+           %Trigger{}
+           |> Trigger.changeset(fix_attrs(Map.put_new(attrs, :start_at, nil)))
+           |> Repo.insert() do
+      if Map.has_key?(attrs, :add_role_ids),
+        do: update_trigger_roles(attrs, trigger),
+        else: {:ok, append_group_labels(trigger)}
+    end
+  end
 
-    handle_action(flow, attrs, 0)
+  @doc """
+  This function is to check the warning while creating the trigger
+  """
+  @spec check_trigger_warnings(map()) ::
+          {:ok, atom()} | {:warning, String.t()}
+  def check_trigger_warnings(attrs) do
+    case Repo.fetch_by(FlowRevision, %{flow_id: Map.get(attrs, :flow_id), status: "published"}) do
+      {:ok, flow} -> handle_action(flow, attrs, 0)
+    end
   end
 
-  @spec handle_action(map(), map(), integer()) :: {:ok, Trigger.t()}
+  @spec handle_action(map(), map(), integer()) :: {:ok, atom()} | {:warning, String.t()}
   defp handle_action(flow, attrs, nested_flow_level) do
     action = flow_action(flow)
 
     case action do
       nil ->
-        do_create_trigger(attrs)
+        {:ok, :no_warnings}
```

Comment:
```suggestion
        :ok
```
Why do we need no warnings atom. Just :ok would work